core/config/toml/toml.go

```diff
@@ -0,0 +1,257 @@
+package toml
+
+import (
+	"net"
+	"net/url"
+
+	ocrnetworking "github.com/smartcontractkit/libocr/networking"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/smartcontractkit/chainlink/core/config"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/ethkey"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/p2pkey"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+//TODO doc
+type CoreConfig struct {
+	// General/misc
+	Dev                 *bool
+	ExplorerURL         *URL
+	InsecureFastScrypt  *bool
+	ReaperExpiration    *models.Duration
+	RootDir             *string
+	ShutdownGracePeriod *models.Duration
+
+	Database *DatabaseConfig
+
+	TelemetryIngress *TelemetryIngressConfig
+
+	Log *LogConfig
+
+	WebServer *WebServerConfig
+
+	// Feeds manager
+	FeatureFeedsManager *bool
+	FeatureUICSAKeys    *bool
+
+	// LogPoller
+	FeatureLogPoller *bool
+
+	// Job Pipeline and tasks
+	JobPipeline *JobPipelineConfig
+
+	// Flux Monitor
+	FMDefaultTransactionQueueDepth *uint32
+	FMSimulateTransactions         *bool
+
+	// OCR V2
+	FeatureOffchainReporting2 *bool
+	OCR2                      *OCR2Config
+
+	// OCR V1
+	FeatureOffchainReporting *bool
+	OCR                      *OCRConfig
+
+	// P2P Networking
+	P2P *P2PConfig
+
+	// Keeper
+	Keeper *KeeperConfig
+
+	// Debugging
+	AutoPprof *AutoPprofConfig
+}
+
+type SecretsConfig struct {
+	DatabaseURL       *URL
+	ExplorerAccessKey string `toml:",omitempty"`
+	ExplorerSecret    string `toml:",omitempty"`
+	//TODO more?
+}
+
+type DatabaseConfig struct {
+	ListenerMaxReconnectDuration  *models.Duration
+	ListenerMinReconnectInterval  *models.Duration
+	Migrate                       *bool
+	ORMMaxIdleConns               *int64
+	ORMMaxOpenConns               *int64
+	TriggerFallbackDBPollInterval *models.Duration
+	// Database Global Lock
+	AdvisoryLockCheckInterval *models.Duration
+	AdvisoryLockID            *int64
+	LockingMode               *string
+	LeaseLockDuration         *models.Duration
+	LeaseLockRefreshInterval  *models.Duration
+	// Database Autobackups
+	BackupDir              *string
+	BackupFrequency        *models.Duration
+	BackupMode             *config.DatabaseBackupMode
+	BackupOnVersionUpgrade *bool
+	BackupURL              *URL
+}
+
+type TelemetryIngressConfig struct {
+	UniConn      *bool
+	Logging      *bool
+	ServerPubKey *string
+	URL          *URL
+	BufferSize   *uint16
+	MaxBatchSize *uint16
+	SendInterval *models.Duration
+	SendTimeout  *models.Duration
+	UseBatchSend *bool
+}
+
+type LogConfig struct {
+	JSONConsole    *bool
+	FileDir        *string
+	Level          *zapcore.Level //TODO is this actually an exceptional case to leave as env var?
```

Comment:
> OTOH it does make it cleaner if everything is in the config file.

Not for test environments or random `func main` scripts that depend on this already.

Reply:
I think I can see how to make this work in the code, but I'm worried that the inconsistency of having the tests continue to support the env var would unecessarily complicate the dev experience. Wdyt about eliminating the runtime reconfiguration of log level and log sql? Pairing that simplification along with only using `LOG_LEVEL` env directly instead of from the toml would makes things more clear IMHO.